model/example_domain_test.go

```diff
@@ -0,0 +1,618 @@
+package model_test
+
+import (
+	"fmt"
+
+	"github.com/nextmv-io/sdk/model"
+)
+
+func ExampleNewDomain_empty() {
+	domain := model.NewDomain()
+
+	fmt.Println(domain.Empty())
+
+	// Output:
+	// true
+}
+
+func ExampleSingleton() {
+	domain := model.Singleton(1)
+
+	fmt.Println(domain.Value())
+
+	// Output:
+	// 1 true
+}
+
+func ExampleMultiple_first() {
+	domain := model.Multiple()
+
+	fmt.Println(domain.Empty())
+
+	// Output:
+	// true
+}
+
+func ExampleMultiple_second() {
+	domain := model.Multiple(3, 5, 1)
+
+	fmt.Println(domain.Min())
+
+	fmt.Println(domain.Max())
+
+	fmt.Println(domain.Len())
+
+	fmt.Println(domain.Contains(3))
+
+	// Output:
+	// 1 true
+	// 5 true
+	// 3
+	// true
+}
+
+/*
+func ExampleNewDomain_singleRange_first() {
+	domain := model.NewDomain(model.NewRange(1, 0))
+
+	fmt.Println(domain.Empty())
+
+	// Output:
+	// true
+}
+*/
+
+func ExampleNewDomain_singleRange_second() {
+	domain := model.NewDomain(model.NewRange(0, 0))
+
+	fmt.Println(domain.Empty())
+
+	fmt.Println(domain.Value())
+
+	fmt.Println(domain.Min())
+
+	fmt.Println(domain.Max())
+
+	iterator := domain.Iterator()
```

Comment:
Commenting here is weird but applies because you are using `Iterator()` here. I believe a test for the `type Iterator` is missing, so it would be good to add that.